plugins/module_utils/network/sonic/facts/snmp/snmp.py

```diff
@@ -0,0 +1,414 @@
+            return view_list
+
+        view_config = snmp_list['vacm']['view']
+
+        for view in view_config:
+            view_list.append({"name": view.get("name"), "included": view.get("include"), "excluded": view.get("exclude")})
+
+        return view_list
+
+    def get_snmp_contact(self, snmp_list):
+        """
+        Get snmp contact from the snmp list
+        """
+        contact_str = ""
+
+        if not snmp_list.get('ietf-snmp-ext:system') or not snmp_list.get('ietf-snmp-ext:system').get('contact'):
+            return contact_str
+
+        snmp_server_config = snmp_list['ietf-snmp-ext:system']['contact']
+
+        if snmp_server_config:
+            contact_str = snmp_server_config
+
+        return str(contact_str)
+
+    def get_snmp_location(self, snmp_list):
+        """
+        Get snmp location from the snmp list
+        """
+        location_str = ""
+        if not snmp_list.get('ietf-snmp-ext:system') or not snmp_list.get('ietf-snmp-ext:system').get('location'):
+            return location_str
+
+        location = snmp_list['ietf-snmp-ext:system']['location']
+
+        if location:
+            return location
+
+        return str(location_str)
+
+    def get_snmp_enable_trap(self, snmp_list):
+        """
+        Get snmp enable trap from the snmp list
+        """
+        enable_trap = list()
+        if not snmp_list.get('ietf-snmp-ext:system'):
+            return enable_trap
+        server = snmp_list.get('ietf-snmp-ext:system')
+
+        if server.get('trap-enable'):
+            enable_trap.append("all")
+        if server.get('notifications'):
+            auth_fail_trap = server.get('notifications').get('authentication-failure-trap')
+            bgp_trap = server.get('notifications').get('bgp-traps')
+            config_change_trap = server.get('notifications').get('config-change-trap')
+            link_down_trap = server.get('notifications').get('link-down-trap')
+            link_up_trap = server.get('notifications').get('link-up-trap')
+            ospf_trap = server.get('notifications').get('ospf-traps')
+
+            if auth_fail_trap:
+                enable_trap.append('auth-fail')
+            elif bgp_trap:
+                enable_trap.append('bgp')
+            elif config_change_trap:
+                enable_trap.append('config-change')
+            elif link_down_trap:
+                enable_trap.append('link-down')
+            elif link_up_trap:
+                enable_trap.append('link-up')
+            elif ospf_trap:
+                enable_trap.append('ospf')
+
+        return enable_trap
+
+    def get_snmp_group(self, snmp_list):
+        """
+        Get snmp group from the snmp list
+        """
+        group_list = list()
+
+        if not snmp_list.get('vacm') or not snmp_list.get('vacm').get('group'):
+            return group_list
+
+        snmp_group_list = snmp_list.get('vacm').get('group')
+
+        for group in snmp_group_list:
+            group = dict()
```

Comment:
This will clear out all content from the input "group" element of the snmp_group_list. Instead, use a different variable name for the "working copy" of the parsed group dictionary. For example:

```suggestion
            group_dict = dict()
```

The additional "suggestions" below are needed to make use of this "working" copy of the group dictionary.

Reply:
Updated.